pkg/util/multilogger/filelogger.go

```diff
@@ -0,0 +1,64 @@
+package multilogger
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/go-logr/stdr"
+	"k8s.io/klog/v2"
+)
+
+var noop = &klog.Logger{}
+
+type fileLogger struct {
+	logger *klog.Logger
+}
+
+func (f *fileLogger) log() *klog.Logger {
+	if f.logger == nil {
+		return noop
+	}
+	return f.logger
+}
+
+func (f *fileLogger) InfoS(msg string, kv ...interface{}) {
+	klog.InfoS(msg, kv...)
+	f.log().Info(msg, kv...)
+}
+
+func (f *fileLogger) ErrorS(err error, msg string, kv ...interface{}) {
+	klog.ErrorS(err, msg, kv...)
+	f.log().Error(err, msg, kv...)
+}
+
+// NewFileLogger creates a logger that writes to both klog and the specified file path.
+// The intention is to be best effort logging, meaning that it will try to create
+// the full file path if it does not exist.
+func NewFileLogger(filePath string) (*fileLogger, error) {
+	err := os.MkdirAll(filepath.Dir(filePath), 0755)
+	if err != nil {
+		return nil, err
+	}
+
+	f, err := os.OpenFile(filePath, os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0644)
```

Comment:
I can't remember how these flags work. Is this going to append to an existing file? Is that what we want for this case? The idea is to get the information about why the most recent pre-run phase failed to emit them as part of the logs in greenboot-healthcheck to speed debugging.

I wonder if we can collect that same information by asking for the journal of the microshift service for the current boot? Something like `journalctl -u microshift --boot`?

Reply:
Oh, good catch, yeah that should be `TRUNC`. In regard to `journalctl`, I like that more, let me see if we can get the same info from that.

Reply:
Changed approach per our discussion.